docs/kb/laptop-dev-env.md

```diff
@@ -0,0 +1,224 @@
+# NovaCustom laptop development environment
+
+This document describes the process of configuring a development and debugging
+environment for NovaCustom laptops.
+
+The instruction is applicable for workstations running Fedora. If you're using
+a different distro, some commands (e.g. package installation) may be different.
+
+## Embedded Controller
+
+This section describes the development environment for EC firmware on NovaCustom
+devices. The Embedded Controller firmware is heavily based on System76 EC
+firmware and makes use of open-source debugging tools and software also
+developed by System76.
+
+### Flashing
+
+This section outlines the parts and software needed to perform external flashing
+of the Embedded Controller.
+
+| Size | 14 inch models | 16 inch models |
+| --- | --- | --- |
+| Programmer | Arduino Mega 2560 | Arduino Mega 2560 |
+| Cable | **FFC 0.5mm pitch 24 pin** | **FFC 1.0mm pitch 24 pin** |
+| Software | ecflash | ecflash |
+
+#### Preparation
+
+1. Obtain a copy of the EC firmware and tools:
+
+    ```bash
+    git clone https://github.com/Dasharo/ec.git
+    cd ec
+    git submodule update --init --checkout
+    ./scripts/deps.sh
+    ```
+
+1. Follow the on-screen instructions to finish configuring your development
+environment.
+1. Connect the Arduino to your workstation with a USB-A to USB-B cable. Do not
+connect the target laptop yet.
+1. Flash the Arduino with debugger firmware:
+
+    ```bash
+    make BOARD=arduino/mega2560
+    make BOARD=arduino/mega2560 flash
+    ```
+
+#### Flash the target laptop
+
+1. In the EC directory, build the EC flashing application:
+
+    ```bash
+    cargo build --manifest-path ecflash/Cargo.toml --example isp --release
+    ```
+
+1. Connect the Arduino to the target laptop with the appropriate FFC cable for
+   your model. If the cable is narrower than the socket on the laptop, align the
+   cable and socket to pin 0 (to the left).
+1. Connect a second cable (e.g. USB-C to USB-C) from the workstation to the
+   target - this cable is needed provide grounding.
+1. Flash the target:
+
+    ```bash
+    sudo ecflash/target/release/examples/isp [path/to/firmware.rom]
+    ```
+
+##### Troubleshooting
+
+In case flashing fails, try the following:
+
+- Try running the command again, sometimes it starts working on the third
+  attempt
+- Try another grounding cable, USB-A to USB-C, or even HDMI to HDMI have been
+  shown to work
+- Try with / without the internal battery connected
+- Try with / without the AC adapter connected
+- Try plugging in the AC adapter right before / after running the command (last
+  resort)
+
+### Building
+
+Build EC firmware:
+
+```bash
+make BOARD=your/board
```

Comment:
I would add a list of laptops and their target BOARD names

Reply:
Done: https://github.com/Dasharo/docs/compare/4e9ec5622c840b8a6d840a400efafdd5de3ee80f..6a19fdc3b9c3ed99815a29ce364e8e90543a5ac2